File: hedera-service/create_topic.js
const dotenv = require("dotenv");
const { Client, PrivateKey, TopicCreateTransaction } = require("@hashgraph/sdk");

dotenv.config();

const { HEDERA_ACCOUNT_ID, HEDERA_PRIVATE_KEY } = process.env;

if (!HEDERA_ACCOUNT_ID || !HEDERA_PRIVATE_KEY) {
  console.error("Missing HEDERA_ACCOUNT_ID or HEDERA_PRIVATE_KEY in .env");
  process.exit(1);
}

async function main() {
  const client = Client.forTestnet();
  const operatorKey = PrivateKey.fromStringECDSA(HEDERA_PRIVATE_KEY);
  client.setOperator(HEDERA_ACCOUNT_ID, operatorKey);

  const txResponse = await new TopicCreateTransaction().execute(client);
  const receipt = await txResponse.getReceipt(client);
  const topicId = receipt.topicId;

  console.log("New Hedera topic created:", topicId.toString());
}

main().catch((error) => {
  console.error("Failed to create topic:", error);
  process.exit(1);
});

File: backend/create_topic.py
import os
from dotenv import load_dotenv
load_dotenv()

from hiero_sdk_python import (
    Client, AccountId, PrivateKey,
    TopicCreateTransaction,
)

account_id = os.getenv("HEDERA_ACCOUNT_ID")
private_key = os.getenv("HEDERA_PRIVATE_KEY")

operatorId = AccountId.from_string(account_id)
operatorKey = PrivateKey.from_string(private_key)

client = Client()
client.set_for_testnet()
client.set_operator(operatorId, operatorKey)

transaction = (
    TopicCreateTransaction(
        memo="SentinelAI Audit Trail"
    )
    .freeze_with(client)
    .sign(operatorKey)
)

receipt = transaction.execute(client)
print(f"Topic created: {receipt.topic_id}")
print(f"View on hashscan: https://hashscan.io/testnet/topic/{receipt.topic_id}")
print(f"SAVE THIS TOPIC ID — add it to your .env file as HEDERA_TOPIC_ID")
